packages/client/README.md

```diff
@@ -116,7 +116,37 @@ OpenFeature.setProvider(new MyProvider())
 In some situations, it may be beneficial to register multiple providers in the same application.
 This is possible using [named clients](#named-clients), which is covered in more detail below.
 
-### Targeting
+### Flag evaluation flow
+
+When a new provider is added to OpenFeature client the following process happens:
+
+```mermaid
+sequenceDiagram
+    autonumber
+    Client-->+Feature Flag Provider: ResolveAll (context)
+    Feature Flag Provider-->-Client: Flags values
+```
+
+In 1 the Client sends a request to the provider backend in order to get all values from all feature flags that it has.
+Once the provider backend replies (2) the client holds all flag values and therefore the flag evaluation process is synchronous.
+
+In order to prevent flag evaluation to the default value while flags are still being fetched it is highly recommended to only lookup for flags value after the provider has emitted the `Ready` event. The following code snipet provides an example.
```

Comment:
```suggestion
In order to prevent flag evaluation to the default value while flags are still being fetched it is highly recommended to only look for flag value after the provider has emitted the `Ready` event. The following code snippet provides an example.
```